jobs/sql/extractions/all_encounters.sql

```diff
@@ -0,0 +1,108 @@
+SET @partition = '${partitionNum}';
+set sql_safe_updates = 0;
+set @next_appt_date_concept_id = CONCEPT_FROM_MAPPING('PIH', 5096);
+
+drop temporary table if exists temp_all_encounters;
+create temporary table temp_all_encounters
+(
+    encounter_id         int,
+    encounter_datetime   datetime,
+    patient_id           int,
+    visit_id             int,
+    creator              int(11),
+    user_entered         varchar(255),
+    encounter_location   varchar(255),
+    encounter_type_name  varchar(50),
+    entered_datetime     datetime,
+    emr_id               varchar(15),
+    next_appt_date       date,
+    voided               bit,
+    last_updated         datetime(3)
+);
+
+-- If there is not a previous watermark, initialize with all encounters
+insert into temp_all_encounters (encounter_id)
+select encounter_id from encounter
+where @previousWatermark is null;
```

Comment:
You may be right about that.  This all makes my head hurt a little, but I think that is consistent with the approach I've described otherwise.

Reply:
Actually, looking at it I think it's ok.  We will always insert newWatermark, so we will always go back to get data changed after newWatermark.  If we happen to load data in that is more recent that this from the source DB, that's fine.  It's not a problem to have more up-to-date data in the target DB, as long as we know that we need to go back and query from the newWatermark value the next time, and not from the datetime of the actual change in the dbevent_patient table.  Make sense?  (This can make one's head hurt)

Reply:
Yep, makes sense... it did seem like *more* up-to-date data wouldn't be a problem, but I wanted to make sure I was understanding correctly.